### ISSUE TITLE
Int16 softmax is using std::vector

### ISSUE THREAD

**Issue by advaitjain** [tensorflow/tensorflow#43126]
@tensorflow/micro

**System information**
- Host OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Debian
- TensorFlow installed from (source or binary): Source
- Tensorflow version (commit SHA if source): 35d9474383c9befd99f457031ada977d681742c4
- Target platform (e.g. Arm Mbed OS, Arduino Nano 33 etc.): STM32F4

**Describe the problem**
This command:
```
make -f tensorflow/lite/micro/tools/make/Makefile TARGET=stm32f4 kernel_softmax_test
```

Fails with the following error message:
```
/home/advaitjain/tensorflow/github/tensorflow/tensorflow/lite/micro/tools/make/downloads/gcc_embedded/bin/../lib/gcc/arm-none-eabi/7.3.1/thumb/v7e-m/libgcc.a(unwind-arm.o): In function `get_eit_entry':
unwind-arm.c:(.text+0x138): undefined reference to `__exidx_end'
unwind-arm.c:(.text+0x13c): undefined reference to `__exidx_start'
collect2: error: ld returned 1 exit status
```

The underlying issue is that the reference implementation of int16 softmax uses std::vector which is incompatible with Micro.

https://github.com/tensorflow/tensorflow/blob/35d9474383c9befd99f457031ada977d681742c4/tensorflow/lite/kernels/internal/reference/softmax.h#L168

The near-term fix will be to revert PR #38873. Longer-term we would have to fix up the reference implementation to be friendly for embedded platforms.

This issue was not caught by continuous integration when the PR was merged because we do not build for TARGET=stm32f4 with only the reference kernels. We only do so with TAGS=cmsis-nn.

Two ways to safeguard against this in the future would be:
 1. build STM32F4 without any additional TAGS
 1. fix up bluepill and add more tests to that target -- this is likely preferable.

**Comment by advaitjain** [comment]
Tagging @freddan80 and @sicong-li-arm 

I'm not thrilled with the revert because of the effort involved in getting this landed from everyone involved. I hope there aren't too many side effects. Let's figure out a proper fix if this is high priority for you guys.

**Comment by freddan80** [comment]
@advaitjain Is this issue resolved with #43180 ?

**Comment by advaitjain** [comment]
> @advaitjain Is this issue resolved with #43180 ?

Yes, we should be able to revert #43127 once #43180 is merged.

**Comment by sicong-li-arm** [comment]
@advaitjain @freddan80 Just came back to office. Sorry for the oversight! The fix looks good to me. Ping me if there's anything I need to do.

**Comment by advaitjain** [comment]
Created #43320 to "revert the revert".
